File: GameLibrary/src/cc/lib/zombicide/ZEquipSlot.java
package cc.lib.zombicide;

import cc.lib.annotation.Keep;
import cc.lib.game.Utils;
import cc.lib.ui.IButton;

@Keep
public enum ZEquipSlot implements IButton {
    LEFT_HAND("LH"),
    BODY("Bo"),
    RIGHT_HAND("RH"),
    BACKPACK("BP");

    ZEquipSlot(String shorthand) {
        this.shorthand = shorthand;
    }

    final String shorthand;

    @Override
    public String getTooltipText() {
        return null;
    }

    @Override
    public String getLabel() {
        return Utils.toPrettyString(name());
    }

    public static ZEquipSlot [] wearableValues() {
        return Utils.toArray(LEFT_HAND, BODY, RIGHT_HAND);
    }
}

File: GameLibrary/src/cc/lib/zombicide/ZEquipSlot.kt
package cc.lib.zombicide

import cc.lib.annotation.Keep

import cc.lib.ui.IButton
import cc.lib.utils.prettify

@Keep
enum class ZEquipSlot(val shorthand: String) : IButton {
    LEFT_HAND("LH"),
    BODY("Bo"),
    RIGHT_HAND("RH"),
    BACKPACK("BP");

    override fun getTooltipText(): String? {
        return null
    }

    override fun getLabel(): String {
        return prettify(name)
    }

    companion object {
        @JvmStatic
        fun wearableValues(): Array<ZEquipSlot> {
            return arrayOf(LEFT_HAND, BODY, RIGHT_HAND)
        }
    }
}